src/plugins/expressions/common/expression_functions/specs/tests/map_column.test.ts

```diff
@@ -0,0 +1,137 @@
+      expect(result.columns[nameColumnIndex].meta).toHaveProperty('type', 'number');
+      expect(result.rows[arbitraryRowIndex]).toHaveProperty('name', 202);
+    });
+  });
+
+  it('adds a column to empty tables', () => {
+    return runFn(emptyTable, { name: 'name', expression: pricePlusTwo }).then((result) => {
+      expect(result.type).toBe('datatable');
+      expect(result.columns).toHaveLength(1);
+      expect(result.columns[0]).toHaveProperty('name', 'name');
+      expect(result.columns[0].meta).toHaveProperty('type', 'null');
+    });
+  });
+
+  it('should assign specific id, different from name, when id arg is passed for new columns', () => {
+    return runFn(emptyTable, { name: 'name', id: 'myid', expression: pricePlusTwo }).then(
+      (result) => {
+        expect(result.type).toBe('datatable');
+        expect(result.columns).toHaveLength(1);
+        expect(result.columns[0]).toHaveProperty('name', 'name');
+        expect(result.columns[0]).toHaveProperty('id', 'myid');
+        expect(result.columns[0].meta).toHaveProperty('type', 'null');
+      }
+    );
+  });
+
+  it('should assign specific id, different from name, when id arg is passed for copied column', () => {
+    return runFn(testTable, { name: 'name', id: 'myid', expression: pricePlusTwo }).then(
+      (result) => {
+        const nameColumnIndex = result.columns.findIndex(({ name }) => name === 'name');
+        expect(result.type).toBe('datatable');
+        expect(result.columns[nameColumnIndex]).toEqual({
+          id: 'myid',
+          name: 'name',
+          meta: { type: 'number' },
+        });
+      }
+    );
+  });
+
+  it('should copy over the meta information from the specified column', () => {
+    return runFn(
+      {
+        ...testTable,
+        columns: [
+          ...testTable.columns,
+          // add a new entry
+          {
+            id: 'myId',
+            name: 'myName',
+            meta: { type: 'date' },
+          },
+        ],
+        rows: testTable.rows.map((row) => ({ ...row, myId: Date.now() })),
+      },
+      { name: 'name', copyMetaFrom: 'myId', expression: pricePlusTwo }
+    ).then((result) => {
+      const nameColumnIndex = result.columns.findIndex(({ name }) => name === 'name');
+      expect(result.type).toBe('datatable');
+      expect(result.columns[nameColumnIndex]).toEqual({
+        id: 'name',
+        name: 'name',
+        meta: { type: 'date' },
+      });
+    });
+  });
+
+  it('should be resilient if the references column for meta information does not exists', () => {
+    return runFn(emptyTable, { name: 'name', copyMetaFrom: 'time', expression: pricePlusTwo }).then(
+      (result) => {
+        expect(result.type).toBe('datatable');
+        expect(result.columns).toHaveLength(1);
+        expect(result.columns[0]).toHaveProperty('name', 'name');
+        expect(result.columns[0]).toHaveProperty('id', 'name');
+        expect(result.columns[0].meta).toHaveProperty('type', 'null');
```

Comment:
Can we add a test here that still infers the type correctly from the first row if the meta information transfer fails?